modules/core/src/test/java/org/apache/ignite/testframework/junits/GridAbstractTest.java

```diff
@@ -864,6 +867,105 @@ protected Ignite startGrid(String igniteInstanceName, IgniteConfiguration cfg, G
     }
 
     /**
+     * Starts new grid of given version and name in separate JVM.
+     *
+     * Uses an ignite-core artifact in the Maven local repository,
+     * if it isn't exists there, it will be downloaded and stored via Maven.
+     *
+     * @param igniteInstanceName Instance name.
+     * @param ver Ignite version.
+     * @param clos IgniteClosure.
+     * @return Started grid.
+     * @throws Exception If failed
+     */
+    protected Ignite startGrid(String igniteInstanceName, String ver,
+        IgniteInClosure<IgniteConfiguration> clos) throws Exception {
+        return startGrid(igniteInstanceName, ver, null, clos, null, true);
+    }
+
+    /**
+     * Starts new grid of given version and name in separate JVM.
+     *
+     * Uses an ignite-core artifact in the Maven local repository,
+     * if it isn't exists there, it will be downloaded and stored via Maven.
+     *
+     * @param igniteInstanceName Instance name.
+     * @param ver Ignite version.
+     * @param cfg Ignite configuration.
+     * @param clos IgniteClosure.
+     * @return Started grid.
+     * @throws Exception If failed
+     */
+    protected Ignite startGrid(String igniteInstanceName, String ver, IgniteConfiguration cfg,
+        IgniteInClosure<IgniteConfiguration> clos) throws Exception {
+        return startGrid(igniteInstanceName, ver, cfg, clos, null, true);
+    }
+
+    /**
+     * Starts new grid of given version and name in separate JVM.
+     *
+     * Uses an ignite-core artifact in the Maven local repository,
+     * if it isn't exists there, it will be downloaded and stored via Maven.
+     *
+     * @param igniteInstanceName Instance name.
+     * @param ver Ignite version.
+     * @param cfg Ignite configuration.
+     * @param clos IgniteClosure for post-configuration.
+     * @param jvmArgs Additional JVM arguments.
+     * @param resetDiscovery Reset DiscoverySpi.
+     * @return Started grid.
+     * @throws Exception If failed.
+     */
+    protected Ignite startGrid(String igniteInstanceName, String ver, IgniteConfiguration cfg,
+        IgniteInClosure<IgniteConfiguration> clos, Collection<String> jvmArgs,
+        boolean resetDiscovery) throws Exception {
+
+        assert !isFirstGrid(igniteInstanceName);
+
+        if (cfg == null)
+            cfg = optimize(getConfiguration(igniteInstanceName));
+
+        Collection<String> filteredJvmArgs = new ArrayList<>();
+
+        filteredJvmArgs.add("-ea");
```

Comment:
Constructor was extended:
https://github.com/daradurvs/ignite/pull/17/files#diff-dbed5523c9851122582415113178aa0aR152
I need an opportunity to pass custom arguments.
Anonymous class with method overriding will complicate the code.
I think constructor extension more clear than anonymous class creation. Don't you?